lamkey_nick_ndvi.py

```diff
@@ -0,0 +1,723 @@
+       )
+date_form = DateFormatter("%b")
+ax.xaxis.set_major_formatter(date_form)
+
+### DO NOT REMOVE LINES BELOW ###
+final_masked_solution = nb.convert_axes(plt, which_axes="current")
+
+
+# In[11]:
+
+
+# Ignore this cell for the autograding tests
+
+
+# In[12]:
+
+
+# Ignore this cell for the autograding tests
+
+
+# # Question 1 (10 points)
+# 
+# Imagine that you are planning NEON’s upcoming flight season to capture remote sensing data in these locations and want to ensure that you fly the area when the vegetation is the most green.
+# 
+# When would you recommend the flights take place for each site? 
+# 
+# Answer the question in 2-3 sentences in the Markdown cell below.
+
+# At the SJER site I'd recommend flying between March and April when NDVI is highest for this site. I would recommend flying in the HARV site during Mid June where NDVI is highest but the following months through October would also be adequate. 
+
+# # Question 2 (10 points)
+# 
+# How could you modify your workflow to look at vegetation changes over time in each site? 
+# 
+# Answer the question in 2-3 sentences in the Markdown cell below.
+
+# Instead of calculating mean NDVI the final outputs could be a series of plots showing the NDVI at each site. I would use the master for loop the same way but store each NDVI calculation as a xarray dataarray and write a loop that plots each xarray in a sequence. Then a visual comparison could be made between each site. 
+
+# # Do not edit this cell! (10 points)
+# 
+# The notebook includes:
+# * additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Do not edit this cell! (20 points)
+# 
+# The notebook will also be checked for overall clean code requirements as specified at the **top** of this notebook. Some of these requirements include (review the top cells for more specifics): 
+# 
+# * Notebook begins at cell [1] and runs on any machine in its entirety.
+# * PEP 8 format is applied throughout (including lengths of comment and code lines).
+# * No additional code or imports in the notebook that is not needed for the workflow.
+# * Notebook is fully reproducible. This means:
+#    * reproducible paths using the os module.
+#    * data downloaded using code in the notebook.
+#    * all imports at top of notebook.
+
+# # BONUS - Export a  .CSV File to Share (10 points possible)
+# 
+# This is optional - if you export a **.csv** file with the columns specified above: Site, Date and NDVI Value you can get an additional 10 points.
+# 
+# * FULL CREDIT: File exists in csv format and contains the columns specified.
+# We will check your github repo for this file!
+# 
+
+# In[13]:
+
+
+output_file = os.path.join(output_dir, "mean_ndvi_df.csv")
+mean_ndvi_df.to_csv(output_file)
+
```

Comment:
@nlamkey I've finished my review of the code. Overall nice job. Like you I was unable to actually check the boxes in the checklist, so added applicable comments inline during the review.